src/webots/nodes/WbConnector.cpp

```diff
@@ -371,12 +371,17 @@ void WbConnector::snapOrigins(WbConnector *other) {
     for (int i = 0; i < 3; ++i)
       h[i] /= 2.0;
   }
-
+// gcc 12.1.0 is raising a false positive warning here about dangling pointers
+#pragma GCC diagnostic push
+#if __GNUC__ == 12 && __GNUC_MINOR__ == 1 && __GNUC_PATCHLEVEL__ == 0
+#pragma GCC diagnostic ignored "-Wdangling-pointer"
+#endif
```

Comment:
The line above is not deleted in the develop branch.
```suggestion

// gcc 12.1.0 is raising a false positive warning here about dangling pointers
#pragma GCC diagnostic push
#if __GNUC__ == 12 && __GNUC_MINOR__ == 1 && __GNUC_PATCHLEVEL__ == 0
#pragma GCC diagnostic ignored "-Wdangling-pointer"
#endif
```